SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -559,10 +559,10 @@ int64 USpatialActorChannel::ReplicateActor()
 	}
 
 	if (SpatialGDKSettings->bEnableUnrealLoadBalancer &&
-		NetDriver->LoadBalanceStrategy != nullptr &&
 		// TODO: the 'bWroteSomethingImportant' check causes problems for actors that need to transition in groups (ex. Character, PlayerController, PlayerState),
 		// so disabling it for now.  Figure out a way to deal with this to recover the perf lost by calling ShouldChangeAuthority() frequently. [UNR-2387]
-		Actor->HasAuthority() &&
+		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID) &&
+		!NetDriver->LockingPolicy->IsLocked(Actor) &&
```

Comment:
In my mind for most use cases and in most of the time we should check the load-balance strategy before the locking policy because it's broader. My line of reasoning is the following - checking LB first prevents a lookup in the lock map pretty much all the time (because most of the time most entities don't need to relinquish authority), on the other hand checking lock policy first prevents position check (currently quite cheap because it's just a based on boxes) when the actor is locked which isn't as good (because most of the time most actors aren't locked). Does this make sense? As far as I'm aware ReplicateActor is on the critical path and we should put extra care to ensure our networking isn't slower than native.

Reply:
Considering that both LB and locking are potentially user-defined functions, I thought it makes the most sense to let the user order them. In that way, I think something like the replication graph pattern or just making LB and locking a composable unit of the same thing makes more sense.

Reply:
> In my mind for most use cases and in most of the time we should check the load-balance strategy before the locking policy because it's broader. 

Nice yea, I agree

> Considering that both LB and locking are potentially user-defined functions, I thought it makes the most sense to let the user order them. In that way, I think something like the replication graph pattern or just making LB and locking a composable unit of the same thing makes more sense.

Sounds like a good FR